sphinx/ext/autosummary/generate.py

```diff
@@ -335,8 +334,10 @@ def get_modules(obj: Any) -> Tuple[List[str], List[str]]:
     ns.update(context)
 
     if doc.objtype == 'module':
+        analyzer = ModuleAnalyzer.for_module(name)
+        attr_docs = analyzer.find_attr_docs()
```

Comment:
Could you generate `analyzer` inside `scanner.scan()` too (without removing it from `get_module_attrs`)? It's no cost to call it because cached.

I'd like to merge `ModuleScanner` class and `get_module_attrs()` in nearly future. So it's better not to the interface of `scan()` method.

Reply:
The both have been incorporated.